src/actions/account.js

```diff
@@ -187,17 +249,17 @@ export const { addAccessKey, addAccessKeySeedPhrase, clearAlert } = createAction
         (accountId, contractId, publicKey, successUrl, title) => defaultCodesFor('account.login', {title})
     ],
     ADD_ACCESS_KEY_SEED_PHRASE: [
-        async (accountId, contractName, publicKey) => {
-            const [walletReturnData] = await Promise.all([
-                wallet.addAccessKey(accountId, contractName, publicKey),
-                sendJson('POST', `${ACCOUNT_HELPER_URL}/account/seedPhraseAdded`, {
-                    accountId,
-                    publicKey,
-                    ...(await wallet.signatureFor(accountId))
-                })
-            ]);
-
-            return walletReturnData;
+        async (accountId, contractName, publicKey, isNew, fundingContract, fundingKey) => {
+            if (isNew) {
+                await wallet.createNewAccount(accountId, fundingContract, fundingKey)
+            }
+            const fullAccess = true;
+            const res = await wallet.addAccessKey(accountId, contractName, publicKey, fullAccess)
+            wallet.postSignedJson('/account/seedPhraseAdded', {
```

Comment:
@mattlockyer missing `await`

Reply:
also I wonder if we should change the order of operation to first update contract helper. So that we don't have silent errors leading to inconsistency (e.g.  here we can add seed phrase key to account, but fail call to contract helper and so not be able to display it in Profile)